### ISSUE TITLE
Methodology: define NEEDL agent working contract and promotion boundaries

### ISSUE THREAD

**Issue by TeaShaman-cyber** [TeaShaman-cyber/theseus-needle-lab#73]
## Motivation

NEEDL already has two complementary authority documents:

- `docs/architecture.md` defines the evidence/authority chain:
  `Issue -> commit/PR -> GitHub Action -> artifact + SHA-256 -> evaluation -> verification receipt -> disposition`.
- `docs/experiment-lifecycle.md` defines the experiment lifecycle:
  `IDEA -> SPECIFIED -> RUNNING -> VERIFYING -> ACCEPTED | REJECTED | INCONCLUSIVE`.

What is still missing is an explicit NEEDL-specific **agent working contract**: the operational methodology that tells autonomous agents how to move through those boundaries without silently promoting execution into evidence, evidence into adjudication, or corpus availability into training authority.

A useful precedent now exists in `theseus-public-observatory/docs/methodology.md`, where an Agent Working Contract defines goal-bound work, roughly five logical actions as an attractor, durable checkpoints, bounded exploration, and natural stopping boundaries. NEEDL needs the same pattern, but with experiment-specific authority semantics.

Recent PR #72 work exposed concrete examples:
- shared operational Session Search storage was initially used for heavy experiment materialization; this should instead use isolated experiment-owned snapshots;
- Codex findings were more useful when grouped into failure classes/invariants than patched one-by-one;
- runtime SHA alone was insufficient without binding clean executable worktree state;
- successful corpus materialization must not imply label, dataset, training, or scientific authority.

## Intended invariant

Add a NEEDL methodology layer that makes these distinctions explicit:

```text
available != authoritative
observed != adjudicated
executed != verified
verified artifact != accepted experiment
candidate corpus != training dataset
training-capable != training-authorized
review finding != required one-off patch
```

The agent contract should encode at least these working invariants:

1. **Experiment-bound work** — each cycle advances one explicit experiment, gate, issue, or verification question.
2. **Five actions as an attractor** — roughly five logical actions per work cycle, not a hard limit.
3. **Durable over conversational** — material decisions, changed invariants, failures, identities, and dispositions land in repository/Issue/receipt/test/checkpoint state.
4. **Execution is not evidence promotion** — successful execution proves only its declared postconditions.
5. **Frozen-input authority** — heavy experiment work consumes immutable experiment-owned snapshots rather than shared operational stores.
6. **Runtime identity is provenance** — bind the executable runtime actually used, not merely a nominal branch or commit.
7. **Observed behavior is evidence, not ground truth** — provider actions/traces do not become `PROBE / READY / UNKNOWN / NO_CALL` or training labels without adjudication.
8. **Training is a protected promotion boundary** — corpus existence, shortlist generation, QA success, or model availability do not authorize training.
9. **Fail closed on semantic ambiguity** — missing mappings, stale provenance, unsupported semantics, path aliasing, or uncertain runtime state remain blocked/degraded/unknown.
10. **Stage -> verify -> promote -> read back** for important derived artifacts.
11. **Review findings map to error classes** — prefer general invariants and regression families over one-off symptom patches.
12. **Resource isolation is part of correctness** — experiment compute/I/O must not interfere with shared operational services.
13. **Natural stopping boundary** — a second independent work unit gets checkpointed rather than silently expanding scope.

## Proposed shape

Prefer a repository-owned `docs/methodology.md` that connects existing architecture and experiment lifecycle rather than creating a standalone manifesto.

The Agent Working Contract should be one section of that methodology, alongside:
- authority / evidence boundaries;
- experiment admission and promotion gates;
- provenance and runtime identity;
- observation vs adjudication;
- training authorization boundary;
- failure / degraded-state semantics.

The Observatory text is a precedent, not a template to copy mechanically.

## Acceptance criteria

- [ ] `docs/methodology.md` exists and references the existing architecture and lifecycle rather than duplicating them.
- [ ] It contains a NEEDL-specific Agent Working Contract.
- [ ] The contract includes “five actions as an attractor” as a workflow heuristic, not a hard execution limit.
- [ ] It explicitly separates execution, verification, adjudication, experiment acceptance, and training authorization.
- [ ] It states that heavy experiment processing must use isolated experiment-owned snapshots rather than shared operational stores.
- [ ] It treats automated/human review findings as signals of error classes/invariants before one-off fixes.
- [ ] It preserves existing Project Contract and workspace-rule precedence.
- [ ] Repository QA covers the durable presence of the key methodological invariants without overfitting tests to exact prose.
- [ ] No training command or new training authority is introduced by this methodology change.

## Non-goals

- Redefining the typed System-1 decision contract from #55.
- Authorizing training.
- Changing current experiment dispositions.
- Copying the Observatory methodology verbatim.
- Turning the working contract into a heavyweight process framework.

**Comment by TeaShaman-cyber** [comment]
## Methodology delta — reciprocal upstream learning

Please include the following NEEDL-specific working principle in the methodology/Agent Working Contract:

```text
upstream is not a static dependency
downstream is not only a consumer

upstream change -> currentness reprobe / possible local issue retirement
downstream exact repro -> possible upstream evidence/test/fix
```

The agent should classify upstream interaction before continuing stale work:

- `UPSTREAM_RESOLVES`
- `UPSTREAM_REFRAMES`
- `DOWNSTREAM_FEEDBACK`
- `FEEDBACK_CANDIDATE`

This layer must remain distinct from `requires` dependencies and from publication authority. A reproducible downstream result may justify preparing upstream feedback, but posting an external issue/PR remains an explicit consequential action.

Concrete current examples are recorded in roadmap #51: current upstream seed support resolves the old #44 seed subproblem; upstream #91 corrected the 2.x causal interpretation; upstream #134 reframes #61/#65 toward deployment-equivalent W4/reference vs engine rather than float-JAX vs engine.
